### ISSUE TITLE
FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0

### ISSUE THREAD

**Issue by kcoderk** [freqtrade/freqtrade#10125]
I run freqai in by docker compose on dry run mode the log showed below
```
2024-04-23 19:15:03   df["do_predict"].iloc[-1] = do_preds[-1]
2024-04-23 19:15:03 /freqtrade/freqtrade/freqai/data_drawer.py:356: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!
2024-04-23 19:15:03 You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.
2024-04-23 19:15:03 A typical example is when you are setting values in a column of a DataFrame, like:
2024-04-23 19:15:03 
2024-04-23 19:15:03 df["col"][row_indexer] = value
2024-04-23 19:15:03 
2024-04-23 19:15:03 Use `df.loc[row_indexer, "col"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.
2024-04-23 19:15:03 
2024-04-23 19:15:03 See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
```

**Comment by xmatthias** [comment]
Please get used to filling out the issue template.  Those who frequently omit the template may have their issues closed without comment.
We're asking some very basic questions - by deleting the issue template, you're explicitly choosing not to answer our questions.

Filling out the template helps reduce our workload, and we greatly appreciate that.

**Comment by kcoderk** [comment]
## Describe your environment

  * Operating system: docker
  * Python Version: python3.12
  * CCXT version: 4.2.82
  * Freqtrade Version: 2024.3
  
Note: All issues other than enhancement requests will be closed without further comment if the above template is deleted or not filled out.

## Describe the problem:

*Explain the problem you have encountered*
I write a reinforcelearning model and a strategy, train ,and after train finished,the program supposed to trade and generate signal,but it give me this error

### Steps to reproduce:

  1. write a reinforcelearning model
  2. training and wait training finished

  
### Observed Results:

  * What happened?
   error occured and I saw no signal 
  * What did you expect to happen?
   just produce signal or something I can see

### Relevant code exceptions or logs

Note: Please copy/paste text of the messages, no screenshots of logs please.

  ```
2024-04-23 19:15:03   df["do_predict"].iloc[-1] = do_preds[-1]
2024-04-23 19:15:03 /freqtrade/freqtrade/freqai/data_drawer.py:356: FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!
2024-04-23 19:15:03 You are setting values through chained assignment. Currently this works in certain cases, but when using Copy-on-Write (which will become the default behaviour in pandas 3.0) this will never work to update the original DataFrame or Series, because the intermediate object on which we are setting values will behave as a copy.
2024-04-23 19:15:03 A typical example is when you are setting values in a column of a DataFrame, like:
2024-04-23 19:15:03 
2024-04-23 19:15:03 df["col"][row_indexer] = value
2024-04-23 19:15:03 
2024-04-23 19:15:03 Use `df.loc[row_indexer, "col"] = values` instead, to perform the assignment in a single step and ensure this keeps updating the original `df`.
2024-04-23 19:15:03 
2024-04-23 19:15:03 See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  ```

**Comment by xmatthias** [comment]
Thanks for bringing this to our attention.

Luckily, it's been fixed in the most recent develop version (more specifically - in #10050).
If you're bothered by this in the current image - best use the latest develop version (on docker, that'll be the image `:develop_freqairl` (or `:develop_freqaitorch`).

Otherwise, this will be fixed with the next release (2024.4).
